.github/workflows/main.yml

```diff
@@ -131,6 +118,47 @@ jobs:
           fi
 
 
+  test_modules:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ 3.6 ]
+        torch-version: [ 1.5.0 ]
+        tensorflow-version: [ 1.15.0 ]
+        modules:
+          - { module: "data_aug", test_file: "tests/forte/processors/data_augment"}
+          - { module: "ir",test_file: "tests/forte/processors/ir/"}
+          - { module: "remote", test_file: "tests/forte/remote_processor_test.py"}
+          - { module: "audio_ext",test_file: "tests/forte/data/readers/audio_reader_test.py"}
+          - { module: "stave",test_file: "tests/forte/data/readers/stave_reader_test.py tests/forte/processors/stave_processor_test.py"}
+          - { module: "models", test_file: "tests/forte/models"}
+          - { module: "wikipedia", test_file: "tests/forte/datasets/wikipedia"}
+          - { module: "nlp",test_file: "tests/forte/processors/subword_tokenizer_test.py tests/forte/processors/pretrained_encoder_processors_test.py"}
+          - { module: "extractor",test_file: "tests/forte/train_preprocessor_test.py forte/data/extractors"}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Cache pip
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('setup.py') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+            ${{ runner.os }}-
+      - name: install forte
+        run: |
+          pip install --use-feature=in-tree-build --progress-bar off .
+      - name: Test modules need extra packages
+        run: |
+          pip install --use-feature=in-tree-build --progress-bar off .[test]
+          pip install --use-feature=in-tree-build --progress-bar off .[${{ matrix.modules.module }}]
+          pip install --progress-bar off pytest==5.1.3 coverage
```

Comment:
now since you don't use `coverage`, you don't need to install coverage. Please really think about it when you are making changes, you have to *understand* what you are doing, not just following instructions.